instrumentation/redis/lib/opentelemetry/instrumentation/redis/patches/client.rb

```diff
@@ -29,11 +31,15 @@ def process(commands) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength, M
             attributes['peer.service'] = config[:peer_service] if config[:peer_service]
             attributes.merge!(OpenTelemetry::Instrumentation::Redis.attributes)
 
+            # Parse commands if we need to derive span attributes from them
+            set_value_size_bytes, parsed_commands = parse_commands(commands) if config[:record_value_size] || (config[:db_statement] != :omit)
+
             unless config[:db_statement] == :omit
-              parsed_commands = parse_commands(commands)
-              parsed_commands = OpenTelemetry::Common::Utilities.truncate(parsed_commands, MAX_STATEMENT_LENGTH)
-              parsed_commands = OpenTelemetry::Common::Utilities.utf8_encode(parsed_commands, binary: true)
-              attributes['db.statement'] = parsed_commands
+              attributes['db.statement'] = OpenTelemetry::Common::Utilities.truncate(parsed_commands, MAX_STATEMENT_LENGTH)
+            end
+
+            if config[:record_value_size] && commands.length == 1 && SET_VALUE_SIZE_COMMANDS.include?(commands[0][0])
```

Comment:
pretty gross 😦

Reply:
Technically, this omits `SET` enclosed in `Redis#queue`, right? Do we want that?

Reply:
This is a few iterations ago - I think we're all set in `sent_value_size` and `retrieved_value_size` now